pandas/core/algorithms.py

```diff
@@ -415,7 +415,7 @@ def unique(values):
 
     >>> pd.unique(pd.array([1 + 1j, 2, 3]))
     <NumpyExtensionArray>
-    [(1+1j), (2+0j), (3+0j)]
+    [np.complex128(1+1j), np.complex128(2+0j), np.complex128(3+0j)]
```

Comment:
I merged https://github.com/pandas-dev/pandas/pull/54268 a while back to make these reprs render like the Python scalars/pre numpy 2.0. It appears that PR didn't touch all the relevant repr methods

Reply:
I'll do a precursor to fix the reprs for arrays, and then revert some of the outputs here.

Your comment only applies to EA reprs?

Reply:
I would say any repr in pandas shouldn't be showing the NEP 51 style repr for scalars.

Reply:
> I would say any repr in pandas shouldn't be showing the NEP 51 style repr for scalars.

The main motivation behind NEP 51 is that the Python numerical types behave differently from the NumPy scalars.

They give a few examples such as

> Python programmers sometimes write obj is True and will surprised when an object that shows as True fails to pass the test. It is much easier to understand this behavior when the value is shown as np.True_.

So I think we **should** be showing the NEP 51 style repr for scalars.

@mroeschke you have approved this PR so I assume that was a typo?